src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java

```diff
@@ -0,0 +1,74 @@
+package org.graylog.integrations.s3.codec;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.graylog.integrations.s3.Configuration;
+import org.graylog.integrations.s3.ContentType;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.graylog.integrations.s3.ContentType.APPLICATION_JSON;
+import static org.graylog.integrations.s3.ContentType.CLOUD_FLARE_LOG;
+import static org.graylog.integrations.s3.ContentType.TEXT_PLAIN;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.mockito.BDDMockito.given;
+
+@RunWith(MockitoJUnitRunner.class)
+public class S3CodecFactoryTest {
+
+    // Code Under Test
+    @InjectMocks
+    private S3CodecFactory cut;
+
+    // Mocks
+    @Mock Configuration mockConfig;
+    @Mock ObjectMapper mockObjectMapper;
```

Comment:
Unused `ObjectMapper`
```suggestion
```

Reply:
I thought this was needed in order for the @InjectMocks annotation to work. Today I Learned that if an appropriate mock object isn't found, @InjectMocks will just inject a null, which is fine for this test.